index.js

```diff
@@ -0,0 +1,316 @@
+/*   _____   _____   _____   _____   ______   _____   _____   ______   */
+/*  |     | |     | |     | |     | |      | |     | |     | |      |  */
+/*  |  |__| |  |  | |  |  | |  |__| '-,  ,-' |  |  | |  |  | '-,  ,-'  */
+/*  |  |__  |  |  | |  |  | |__   |   |  |   |     | |  |  |   |  |    */
+/*  |  |  | |  |  | |  |  | |  |  |   |  |   |  |  | |  |  |   |  |    */
+/*  |_____| |_____| |__|__| |_____|   |__|   |__|__| |__|__|   |__|    */
+
+//. <a href="https://github.com/fantasyland/fantasy-land"><img alt="Fantasy Land" src="https://raw.githubusercontent.com/fantasyland/fantasy-land/master/logo.png" width="75" height="75" align="left"></a>
+//.
+//. # sanctuary-constant
+//.
+//. A value of type `Constant a b` always contains exactly one value,
+//. of type `a`. Mapping over a `Constant a b` has no effect because
+//. the `b -> c` function is never applied.
```

Comment:
I don't really understand the Constant type, why it takes a type representative, and how that type representative relates to Constant's the type parameters. If you care to explain it to me, maybe that explanation can be added here to the introduction. :)

Reply:
In Haskell, we can do this:

```haskell
> import Data.Functor.Constant

> pure 42 :: Constant String Int
Constant ""
```

We provided `42 :: Int` in “`b`” position; `"" :: String` materialized in “`a`” position.

In JavaScript, keeping track of `a` and `b` in the type system is not an option so we must provide type information explicitly:

```javascript
> S.of (Constant (String)) (42)
Constant (String) ('')
```

Reply:
> `"" :: String` materialized

Woah. o.o .... Ahh, because String is a Monoid. So `Constant (Boolean)` will throw when instantiated.

Alright. That makes things clearer. I don't really see how this would be used in practice, and that's probably what most of my confusion stems from.